### PR TITLE
Make support for nscd flushing optional

### PR DESCRIPTION
**Pull request by keszybz**
Fedora will deprecate support for nscd in the upcoming release [1] and plans to
drop it in the next one [2]. At that point we might as well build systemd
without that support too, since there'll be nothing to talk too.

[1] https://fedoraproject.org/wiki/Changes/DeprecateNSCD
[2] https://fedoraproject.org/wiki/Changes/RemoveNSCD

### CODE REVIEW

File: src/machine/machined.h [hunk 1]
```diff
@@ -57,7 +59,11 @@ int manager_unref_unit(Manager *m, const char *unit, sd_bus_error *error);
 int manager_unit_is_active(Manager *manager, const char *unit);
 int manager_job_is_active(Manager *manager, const char *path);
 
+#if ENABLE_NSCD
 int manager_enqueue_nscd_cache_flush(Manager *m);
+#else
+static inline void manager_enqueue_nscd_cache_flush(Manager *m) {}
```
Comment (bluca) [inline src/machine/machined.h:65]: Will the compiler nag about \*m being unused with the flags we set in Meson?

**Comment by bluca** [comment]
bionic-i386 timed out as usual. Error in TEST-02-UNITTESTS on CentOS is unrealted too, although slightly worrying:

```
08:13:14 --- test-journal-flush begin ---
08:13:14 journal_file_copy_entry failed: Bad message
08:13:14 Assertion 'r >= 0' failed at src/journal/test-journal-flush.c:52, function main(). Aborting.
```

maybe a bad disk write?

**Comment by mrc0mmand** [comment]
> bionic-i386 timed out as usual. Error in TEST-02-UNITTESTS on CentOS is unrelated too, although slightly worrying:
> 
> ```
> 08:13:14 --- test-journal-flush begin ---
> 08:13:14 journal_file_copy_entry failed: Bad message
> 08:13:14 Assertion 'r >= 0' failed at src/journal/test-journal-flush.c:52, function main(). Aborting.
> ```
> 
> maybe a bad disk write? Paging @mrc0mmand

I noticed that too in several cases over the past several days and it's indeed worrying, since it's failing in the CentOS 7 job (i.e. without nested KVM, just regular KVM). I'm not sure if anything has changed in the journal stuff to cause this though...

File: src/machine/machined.h (hunk 1 above)
Comment (keszybz) [reply src/machine/machined.h:65]: I don't think so. With gcc in F33 and rawhide, it doesn't. (And if it turns out it does with some combination of flags, we can address the issue then.)

Comment (bluca) [reply src/machine/machined.h:65]: Great, thanks for checking!

**Pull request merged**